src/Main.elm

```diff
@@ -1,47 +1,165 @@
 module Main exposing (..)
 
 import Browser
-import Html exposing (Html, text, div, h1, img)
-import Html.Attributes exposing (src)
+import Html exposing (..)
+import Html.Attributes exposing (class, href)
+import Html.Events exposing (..)
+import Http
+import Json.Decode as Decode exposing (Decoder, andThen, field, int, list, string, succeed)
+import Json.Decode.Pipeline exposing (required)
+import RemoteData exposing (RemoteData, WebData)
+
+
 
 
 ---- MODEL ----
 
 
+type alias StoryId =
+  Int
+
+type alias Story =
+    { title: String
+    , url: String
+    }
 type alias Model =
-    {}
+    { stories : WebData (List Story)
+    , storyIds : WebData ( List StoryId)
+    }
+
+
+emptyModel : Model
+emptyModel =
+    { stories = RemoteData.Loading
+    ,storyIds = RemoteData.Loading
+    }
 
 
 init : ( Model, Cmd Msg )
 init =
-    ( {}, Cmd.none )
+    ( emptyModel, Cmd.none )
 
 
 
 ---- UPDATE ----
 
 
 type Msg
-    = NoOp
+    = HandleFetchedStoryIds (WebData (List StoryId))
+    | HandleFetchedStory ( WebData (List Story))
 
+newsDecoder : Decoder Story
+newsDecoder =
+    Decode.succeed Story
+        |> required "title" string
+        |> required "url" string
 
-update : Msg -> Model -> ( Model, Cmd Msg )
-update msg model =
-    ( model, Cmd.none )
+
+getStory : Int -> Cmd Msg
+getStory id =
+    Http.get
+        { url = "https://hacker-news.firebaseio.com/v0/item/" ++ ( String.fromInt id ) ++ ".json"
+
+        , expect =
+            list newsDecoder
+                |> Http.expectJson ( RemoteData.fromResult >> HandleFetchedStory )
+        }
+
+getStoryId : Cmd Msg
+getStoryId =
+    Http.get
+        { url = "https://hacker-news.firebaseio.com/v0/topstories.json"
+        , expect =
+            list int
+                |> Http.expectJson (RemoteData.fromResult >>  HandleFetchedStoryIds  )
+        }
 
 
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        HandleFetchedStoryIds id ->
+            let
+                _= Debug.log "hello" id
```

Comment:
@anvmn 
Here I am able to log the list of the Ids and I am wondering if I want to map through the List of id. What variable will I use ( which one contains the list) `List.map "---this variable---"`

Reply:
@Diama1 
So, first thing to node that what you have here is not a `List`, but `WebData (List StoryId)`
So, you'll need to extract the list from within the `RemoteData` before you can map it.

Now the answer to the question. `List.map` signature is:
`map : (a -> b) -> List a -> List b`

So, it's first variable is a function that performs the mapping.
In our case it will be `StoryId` -> `Cmd Msg`.
Note that you have this already - `getStory : Int -> Cmd Msg`, though proper name for it is `fetchStoriesIdsCmd`.